From 70bbb6bb623f9e03fc9462a74fadf57cb15318b5 Mon Sep 17 00:00:00 2001
From: Christian Bruel <christian.bruel@foss.st.com>
Date: Thu, 4 Jan 2024 17:30:56 +0100
Subject: [PATCH 0924/1141] dt-bindings: pci: stm32: Describe wake-gpios for RC

Add binding documentation for optional wake-gpios on pcie nodes

Signed-off-by: Christian Bruel <christian.bruel@foss.st.com>
Change-Id: I7f722f8dbf67f3a18859972b593a5d337ef7e53c
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/350787
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../bindings/pci/st,stm32-pcie-host.yaml         | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

--- a/Documentation/devicetree/bindings/pci/st,stm32-pcie-host.yaml
+++ b/Documentation/devicetree/bindings/pci/st,stm32-pcie-host.yaml
@@ -89,6 +89,19 @@ properties:
   feature-domains:
     maxItems: 1
 
+  wake-gpios:
+    description: GPIO controlled connection to WAKE# input signal
+    maxItems: 1
+
+  wakeup-source: true
+
+if:
+  required:
+     - wakeup-source
+then:
+  required:
+     - wake-gpios
+
 required:
   - st,syscfg
   - interrupt-map
@@ -105,6 +118,7 @@ unevaluatedProperties: false
 
 examples:
   - |
+    #include <dt-bindings/gpio/gpio.h>
     #include <dt-bindings/phy/phy.h>
     #include <dt-bindings/clock/stm32mp25-clks.h>
     #include <dt-bindings/interrupt-controller/arm-gic.h>
@@ -139,5 +153,7 @@ examples:
              phy-names = "pcie-phy";
              resets = <&rcc PCIE_R>;
              reset-names = "pcie";
+             wakeup-source;
+             wake-gpios = <&gpioh 5 GPIO_ACTIVE_LOW>;
              msi-parent = <&v2m0>;
     };
